### ISSUE TITLE
Allow pulses to send entire dashboards

### ISSUE THREAD

**Issue by mattarderne** [metabase/metabase#10022]
### Feature request

I've had a look, and it seems this request doesn't exist (IMO surprisingly). 

I'd like to set a schedule to email dashboards as a PDF. 
This is in order to push reports out to the relevant stakeholders. I realise that it is possible to achieve this with multiple pulses of questions, but feel that a "weekly summary" type email is more manageable.

I've achieved this before in [Zoho Reports](https://www.zoho.com/analytics/help/export/email.html)

Thanks

:arrow_down: Please click the :+1: reaction. Do **not** leave a `+1` or `update?` or `I need this too` comment

**Comment by flamber** [comment]
Hi @mattarderne 
This would require Metabase to have a browser rendering engine, which is unlikely, but never say never.
Kinda a duplicate of #3770 - or at least it would depend on that issue.
A current workaround would be to create a script that uses a headless browser (Puppeteer etc) to take a "screenshot" (PNG/PDF) of the dashboard and then send that.

**Comment by mattarderne** [comment]
Ok thanks, will take a look at that.

**Comment by mattarderne** [comment]
In case someone else would like to use this, I've setup a quick example of the export to pdf  /png below. 

install [Puppeteer](https://github.com/GoogleChrome/puppeteer) with:
```
npm i puppeteer
```
copy  below into metabase.js file:

```javascript
/**
 * @name metabase
 *
 * @desc Logs into metabase. Provide your username and password as environment variables when running the script, i.e:
 * `META_USER=email_address META_PWD=password node metabase.js && open meta.pdf`
 *
 */
const puppeteer = require('puppeteer')
const screenshot = '';
(async () => {
  // const browser = await puppeteer.launch({headless: false}) use this instead of below to troubleshoot
  const browser = await puppeteer.launch();
  const page = await browser.newPage()
  await page.setViewport({ width: 1920, height: 1080});
  console.log('=====Login')
  await page.goto('localhost:3000')
  await page.waitFor(500)
  await page.type('#root > div > div > div > div.Login-content.Grid-cell > form > div:nth-child(3) > input', process.env.META_USER)
  await page.type('#root > div > div > div > div.Login-content.Grid-cell > form > div:nth-child(4) > input', process.env.META_PWD)
  await page.click('#root > div > div > div > div.Login-content.Grid-cell > form > div.Form-actions.p4 > button > div > div')
  await page.waitForNavigation()
  console.log('=====Navigate')
  await page.goto('localhost:3000/dashboard/36', {waitUntil: 'networkidle2'}); 
  await page.waitFor(200)
  console.log('=====Capture')
  await page.pdf({path: 'meta.pdf'})
  await page.screenshot({ path: screenshot })
  browser.close()
  console.log('See screenshot: ' + screenshot)
})()
```

run with following command line:
```shell
META_USER=email_address META_PWD=password node metabase.js && open meta.pdf
```

Notes:
* Metabase blocks multiple logins, so need to improve the above for multiple dashboards etc
* Use `const browser = await puppeteer.launch({headless: false})` to troubleshoot
* Replace `localhost:3000/dashboard/36` with a valid URL to the target

**Comment by exabrial** [comment]
Try using Chromedriver, you'd get better renderings:

```
WebDriver driver = new FirefoxDriver();
driver.get("http://www.google.com/");
File scrFile = ((TakesScreenshot)driver).getScreenshotAs(OutputType.FILE);
FileUtils.copyFile(scrFile, new File("c:\\tmp\\screenshot.png"));
```

**Comment by Gavin1Sinai** [comment]
Really need this feature it will be huge for us

**Comment by PequiAtomico** [comment]
I really need this. Also, the possibility to write messages would be great.

**Comment by bizzareus** [comment]
Badly need this feature

**Comment by chitransh89** [comment]
I have created a Python script to take screenshot using pypeteer (pythonic version of pupeteer which is shared above). It uses the metabase session api instead of entering the username and password via type function - cleaner way of doing things. 

I have hosted the script at
https://gist.github.com/chitransh89/664444c994e9d36d582b44255349d8eb

**Comment by mattarderne** [comment]
Seems like this has been added as a feature? Anyone confirm this? 
https://github.com/metabase/metabase/issues/14575

**Comment by chitransh89** [comment]
Not until 0.39.1 community edition. There was a demand from our client to share dashboard as an image in mail for people in their management. Pulses had their limitations in terms of variety of charts and whatever visual we got on mail wasn't looking nice and correct either (all horizontal bars became of same size)

**Comment by paramaggarwal** [comment]
I just want a way to quickly add all the questions inside a dashboard to also be added to a Pulse email report. Is that possible?

**Comment by flamber** [comment]
@paramaggarwal That's already supported with Dashboard Subscriptions since 0.38.0: https://www.metabase.com/docs/latest/users-guide/dashboard-subscriptions.html

**Comment by chitransh89** [comment]
If you want to go freeway, that's the script for you. Pulse email report
doesn't give much options in terms of charts. And the charts which we are
receiving are not having a nice display, they seem shrinked. Hence we went
the headless browser way.

**Comment by mazameli** [comment]
@flamber Are we keeping this feature request open because of the PDF aspect of the request, or as a way of acknowledging that not all static visualizations match the original visualization still?

**Comment by flamber** [comment]
@mazameli We can close this, since we're likely never going to match dashboard look exactly without having a browser engine.
We have #5493 (catch-all) for the difference in static viz.
